src/toxcore/crypto_core.rs

```diff
@@ -226,6 +236,32 @@ impl FromBytes for sha512::Digest {
 pub mod tests {
     use super::*;
 
+    use std::thread;
+
+    // crypto_init()
+
+    #[test]
+    // test whether crypto init behaves well when one tries to run it in parallel.
+    // Its code should **not** be ran in parallel.
+    fn crypto_init_test() {
```

Comment:
Do we need this test? It looks like it should be tested/guaranteed by sodiumoxide.

Reply:
No it's not. That feature was removed right before release.

See
 https://github.com/sodiumoxide/sodiumoxide/pull/336
 https://github.com/sodiumoxide/sodiumoxide/blob/master/src/lib.rs#L93

Reply:
I know, and that's not what I meant. This test is meant to check thread safety for `crypto_init` but it's not our responsibility to make it thread safe. It should be guaranteed by sodiumoxide,

Reply:
I see. You are right